### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by adavydow**
1. Type fixes
2. Slices implemented to comply with Sequence interface.
3. Fields of the dataset are now read-only properties.
4. Style fixes

### CODE REVIEW

File: face_api_dataset/dataset.py [hunk 1]
```diff
@@ -366,37 +372,72 @@ def transform(modalities: List[Modality]) -> List[Modality]:
...
+
+    @overload
+    def __getitem__(self, index: int) -> dict:
+        pass
+
+    @overload
+    def __getitem__(self, index: slice) -> "FaceApiDataset":
+        pass
+
+    def __getitem__(self, i: Union[int, slice]) -> Union[dict, "FaceApiDataset"]:
```
Comment (Rauf-Kurbanov) [inline face_api_dataset/dataset.py:414]: I'm not a big fan of else statements after a return. But dealer's choice

File: face_api_dataset/dataset.py [hunk 2]
```diff
@@ -366,37 +372,72 @@ def transform(modalities: List[Modality]) -> List[Modality]:
...
         if self._needs_info:
-            info_file = self.root / f"{number}.{_Extension.INFO}"
+            info_file = self._root / f"{number}.{_Extension.INFO}"
             with info_file.open("r") as f:
                 info = json.load(f)
 
         ret = {}
-        for modality in self.modalities:
+        for modality in self._modalities:
             ret[modality] = self._open_modality(modality, number, info)
```
Comment (Rauf-Kurbanov) [inline face_api_dataset/dataset.py:441]: I would rather store `open` logic within the modality object in case we desire to extend them later

**Review by Rauf-Kurbanov — APPROVED** [review]
lgtm, minor commebts

File: face_api_dataset/dataset.py (hunk 2 above)
Comment (adavydow) [reply face_api_dataset/dataset.py:441]: I didn't want to expose modality logic to the user, so I decided to go with a single enum.
I'm thinking about classes, it looks more extensible and probably the way to go in the future, however, providing a list of class names looks a bit worse than a list of enums.

File: face_api_dataset/dataset.py (hunk 1 above)
Comment (adavydow) [reply face_api_dataset/dataset.py:414]: Usually, I'm following this logic:
  Some exceptional situation or short-cut in logic -> no else statement.
  Choice from two (or more) possibilities -> else statement.

File: face_api_dataset/dataset.py (hunk 2 above)
Comment (adavydow) [reply face_api_dataset/dataset.py:441]: Probably enum of classes is the way to go =)

**Pull request merged**